pngcanvas.py

```diff
@@ -258,26 +280,31 @@ def load(self, f):
             raise ValueError('Unsupported PNG format (filter_type={}; must be 0)'.format(filter_type))
         if interlace != 0:
             raise ValueError('Unsupported PNG format (interlace={}; must be 0)'.format(interlace))
-        if color_type != 6:
-            raise ValueError('Unsupported PNG format (color_type={}; must be 6)'.format(color_type))
+        if color_type not in SUPPORTED_COLOR_TYPES:
+            raise ValueError('Unsupported PNG format (color_type={}; must one of {})'.format(SUPPORTED_COLOR_TYPES))
 
         self.width = width
         self.height = height
         self.canvas = bytearray(self.bgcolor * width * height)
-        row_size = width * 4
-        step_size = 1 + row_size
+        bytes_per_pixel = SAMPLES_PER_PIXEL[color_type]
+        bytes_per_row = bytes_per_pixel * width
+        bytes_per_rgba_row = SAMPLES_PER_PIXEL[COLOR_TYPE_TRUECOLOR_WITH_ALPHA] * width
```

Comment:
Oh, now I understand why you need both. I missed a few things in the previous version. You may want to add a comment here saying "We need the RGBA length regardless since we cast RGB to RGBA".